File: fix_work3.js
const fs = require('fs');
const path = require('path');
const htmlPath = path.join(__dirname, 'index.html');
const base64Path = path.join(__dirname, 'replace_image3.txt');
let html = fs.readFileSync(htmlPath, 'utf8');
const base64 = fs.readFileSync(base64Path, 'utf8').replace(/\r?\n/g, '');

const linkPattern = /\s*<a class="work-link" href="mmexport1784018485068\.\.mp4" target="_blank" rel="noopener">观看视频案例<\/a>\s*/;
if (!linkPattern.test(html)) throw new Error('video link not found in first card');
html = html.replace(linkPattern, '');

const imgPattern = /<img src="data:image\/svg\+xml;base64,[^"]*" alt="作品封面 3" \/>/;
if (!imgPattern.test(html)) throw new Error('third card placeholder image not found');
html = html.replace(imgPattern, `<img src="data:image/png;base64,${base64}" alt="作品封面 3" />`);

const insertPattern = /(<img src="data:image\/png;base64,[^"]*" alt="作品封面 3" \/>[\s\S]*?<ul class="tag-list">[\s\S]*?<\/ul>\s*)(<\/div>\s*<\/article>)/;
if (!insertPattern.test(html)) throw new Error('third work card block not found for insertion');
html = html.replace(insertPattern, `$1            <a class="work-link" href="mmexport1784018485068..mp4" target="_blank" rel="noopener">观看 10 分钟视频</a>\n$2`);

fs.writeFileSync(htmlPath, html, 'utf8');
console.log('updated');

File: fix_work3.py
import re
from pathlib import Path

base64_file = Path('replace_image3.txt')
html_file = Path('index.html')
html = html_file.read_text(encoding='utf-8')
base64_data = base64_file.read_text(encoding='ascii').strip()

removed, count = re.subn(
    r"\s*<a class=\"work-link\" href=\"mmexport1784018485068\.\.mp4\" target=\"_blank\" rel=\"noopener\">观看视频案例</a>\s*\n",
    '',
    html,
    count=1,
)
if count == 0:
    raise SystemExit('video link not found in first card')
html = removed

html, count = re.subn(
    r'(<img src=")data:image/svg\+xml;base64,[^\"]+(" alt="作品封面 3" />)',
    lambda m: m.group(1) + 'data:image/png;base64,' + base64_data + m.group(2),
    html,
    count=1,
)
if count == 0:
    raise SystemExit('third card placeholder image not found')

if '观看 10 分钟视频' in html:
    raise SystemExit('target video link already present')

pattern = re.compile(
    r'(<img src="data:image/png;base64,[^"]+" alt="作品封面 3" />.*?<ul class="tag-list">.*?</ul>\n)(\s*</div>\n\s*</article>)',
    re.S,
)
match = pattern.search(html)
if not match:
    raise SystemExit('third work card block not found')

insert_at = match.end(1)
html = (
    html[:insert_at]
    + '            <a class="work-link" href="mmexport1784018485068..mp4" target="_blank" rel="noopener">观看 10 分钟视频</a>\n'
    + html[insert_at:]
)

html_file.write_text(html, encoding='utf-8')
print('updated')